v0/destinations/facebook_offline_conversions/utils.js

```diff
@@ -221,15 +223,73 @@ const getStandardEventsAndEventSetIds = (destination, event) => {
 };
 
 /**
- * Returns content type
- * @param {*} message
- * @returns
+ *
+ * @param {*} message Rudder Payload
+ * @param {*} standardEvent
+ * @param {*} categoryToContent [ { from: 'clothing', to: 'product' } ]
+ * We will be mapping properties.category to user provided content else taking the default value as per ecomm spec
+ * If category is clothing it will be set to ["product"]
+ * @return Content Type array as defined in:
+ * - https://developers.facebook.com/docs/facebook-pixel/reference/#object-properties
  */
-const getContentType = message => {
-  let contentType = "product";
-  const { event } = message;
-  if (event === "Product List Viewed") {
-    contentType = "product_group";
+const getContentType = (message, standardEvent, categoryToContent) => {
+  const { integrations } = message;
+  if (
+    integrations &&
+    integrations.FacebookOfflineConversions &&
+    isObject(integrations.FacebookOfflineConversions) &&
+    integrations.FacebookOfflineConversions.contentType
+  ) {
+    return integrations.FacebookOfflineConversions.contentType;
+  }
+
+  let contentType;
+
+  let { category } = message.properties;
+  // category field is mapped from properties.category, if not found, then, category of the first product item from the products array ( if any )
+  if (!category) {
+    const { products } = message.properties;
+    if (products && products.length > 0 && Array.isArray(products)) {
+      if (isObject(products[0])) {
+        category = products[0].category;
+      }
+    }
+  }
+
+  /** *
+   * if none of the above is followed, we are defaulting it to "product" except whatever events that are mapped to viewContent
+   * for viewContent if there is any product array we are setting content_type as "product_group"
+   * when no product array is not found we are defaulting it to "product"
+   */
+  if (category) {
+    const categoryToContentMapping = getHashFromArray(
+      categoryToContent,
+      "from",
+      "to",
+      false
+    );
+
+    const keys = Object.keys(categoryToContentMapping);
+
+    keys.forEach(key => {
+      if (key === category) {
+        contentType = categoryToContentMapping[key];
+      }
+    });
+
+    // if category isn't mapped with contentType then keeping default contentType as "product"
+    if (!contentType) {
+      contentType = "product";
+    }
+  } else if (standardEvent === "ViewContent") {
+    const { products } = message.properties;
+    if (products && products.length > 0 && Array.isArray(products)) {
```

Comment:
This will be opposite. if product array is present, then mapped to "`product`", else mapped to "`product_group`".

Reply:
and also same thing happens, when the event name is `product list viewed`